### DISCUSSION TITLE
Does autogluon time series have automatic feature engineering?

### DISCUSSION THREAD

**Discussion by isCopyman** [autogluon/autogluon#4077] [Q&A]
It seems that there is no automatic feature engineering in the documents and tutorials

**Comment by shchur** [comment]
Currently, automatic feature engineering in AutoGluon-TimeSeries is limited to automatically selecting lags/generating time features (e.g., day of week, month, etc) for tabular & deep learning models. You can change these by overriding the model hyperparameters listed here https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-model-zoo.html.

**Reply by isCopyman** [reply]
Will autogluon-timeseries automatically perform normalization and de normalization？

**Reply by shchur** [reply]
Yes, AutoGluon automatically applies normalization & denormalization for models where this is beneficial for the performance (e.g., models like DeepAR, LightGBM, TFT, PatchTST).

**Reply by krds00** [reply]
> Currently, automatic feature engineering in AutoGluon-TimeSeries is limited to automatically selecting lags/generating time features (e.g., day of week, month, etc) for tabular & deep learning models. You can change these by overriding the model hyperparameters listed here https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-model-zoo.html.

could you explain, please, how to extract feature importance of generated lags features? As I see, now we can't

**Comment by rahuljungbahadur** [comment]
You can use this function to extract the feature importances from the tabular model.

```
import pickle

def get_model_ft_imp(path_to_model_pkl:str):

    ## Load all components of the model 
    model_obj = []
    with (open(model_path, "rb")) as openfile:
        while True:
            try:
                model_obj.append(pickle.load(openfile))
            except EOFError:
                break
    ## Extract the model
    ag_tab_model = model_obj[0]

    ## extract the model
    tab_model = ag_tab_model.model.dump_model()


    ## get feature importances
    return tab_model['feature_importances']
    
```

Just paste the path_to_model_pkl based on where exactly the model.pkl for the DirectTabular is saved
